src/main-app.js

```diff
@@ -174,6 +260,42 @@ export class MainApplication extends LitElement {
     }
   }
 
+  async _startScan() {
+    const scanOption = { recordType: "mime" };
+    try {
+      if (this._btnScanSvg.classList.contains("disabled-scan")) {
+        await this._reader.scan(scanOption);
+        this._btnScanSvg.classList.remove("disabled-scan");
+        this._snackbar.labelText = "Add item or touch an NFC tag!";
+        this._actionBtn.textContent = "";
+        this._snackbar.open();
+        if (this._firstScan) {
+          this._hideScanBg = true;
+          this._firstScan = false;
+          this.requestUpdate();
+        }
+      } else {
+        const controller = new AbortController();
+        scanOption.signal = controller.signal;
+        // New invocation of scan() will cancel previous scan().
+        await this._reader.scan(scanOption);
+        controller.abort();
+        this._btnScanSvg.classList.add("disabled-scan");
+        this._snackbar.labelText = "NFC tags scan operation is disabled!";
```

Comment:
We should probably use . instead of ! in most places

Reply:
Done.